src/components/dialogs/explicit-naming-filter-creation-dialog.js

```diff
@@ -0,0 +1,196 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import DialogTitle from '@mui/material/DialogTitle';
+import DialogContent from '@mui/material/DialogContent';
+import DialogActions from '@mui/material/DialogActions';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import React, { useEffect, useRef, useState } from 'react';
+import makeStyles from '@mui/styles/makeStyles';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { createFilter, saveFilter } from '../../utils/rest-api';
+import { FilterType } from '../../utils/elementType';
+import { useSelector } from 'react-redux';
+import Alert from '@mui/material/Alert';
+import PropTypes from 'prop-types';
+
+import ExplicitNamingFilterDialogContent from './explicit-naming-filter-dialog-content';
+
+const useStyles = makeStyles((theme) => ({
+    dialogPaper: {
+        width: 'auto',
+        minWidth: '800px',
+        minHeight: '400px',
+        margin: 'auto',
+    },
+}));
+
+const ExplicitNamingCreationDialog = ({
+    id,
+    open,
+    onClose,
+    name,
+    title,
+    isFilterCreation,
+}) => {
+    const intl = useIntl();
+    const classes = useStyles();
+    const [isGeneratorOrLoad, setIsGeneratorOrLoad] = useState(false);
+    const [equipmentType, setEquipmentType] = useState(null);
+
+    const [createFilterErr, setCreateFilterErr] = React.useState('');
+    const activeDirectory = useSelector((state) => state.activeDirectory);
+
+    const [tableValues, setTablesValues] = useState([]);
+
+    const fetchFilter = useRef(null);
+    fetchFilter.current = open && !isFilterCreation;
+
+    const resetDialog = () => {
+        setEquipmentType('');
+        setCreateFilterErr('');
+    };
+
+    useEffect(() => {
+        setIsGeneratorOrLoad(
+            equipmentType === 'GENERATOR' || equipmentType === 'LOAD'
+        );
+    }, [equipmentType]);
+
+    const handleEditCallback = (
+        tableValues,
+        isGeneratorOrLoad,
+        isFilterCreation,
+        equipmentType,
+        name,
+        id
+    ) => {
+        setEquipmentType(equipmentType);
+        setTablesValues(tableValues);
+    };
+
+    useEffect(() => {
+        setCreateFilterErr('');
+    }, [tableValues]);
+
+    const handleCreateFilter = () => {
+        if (
+            tableValues.every((el) => {
+                if (!el?.equipmentID) {
```

Comment:
why not directly use if(tableValues.every((el) => !el?.equipmentID)) ? this will check every element has equipment id

Reply:
here i need to check every id else i display error message

Reply:
I don't like this mixing of 'every' with just 'last'.
setCreateFilterErr is a side-effect.
It should not be call in the expressions of the 'if'.
Please call this side effect in its own branch of the if.
A way to do so is
```
const hasMissingId = tableValues.some((el) => !el?.equipmentID);
if (hasMissingId) {
  setCreateFilterErr(
      intl.formatMessage({
          id: 'missingEquipmentsIdsError',
      })
  );
} else {
  // your previous 'then'
}
```
Of course you can inline hasMissingId, if you prefer.

Reply:
ok i changed like this